### PR TITLE
Codeblock Dev Language Missing Validation add LineCount

### PR DESCRIPTION
**Pull request by xichan-ms**
Codeblock Dev Language Missing Validation add LineNum

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/6421)

### CODE REVIEW

**Comment by xichan-ms** [comment]
AB#205281

File: src/docfx/lib/markdown/validation/DocsValidationExtension.cs [hunk 1]
```diff
@@ -133,6 +133,7 @@ internal static class DocsValidationExtension
                     codeBlockItem.Info = fencedCodeBlock.Info;
                     codeBlockItem.Arguments = fencedCodeBlock.Arguments;
                     codeBlockItem.IsOpen = fencedCodeBlock.IsOpen;
+                    codeBlockItem.LineNum = GetFencedCodeBlockNetLineNumber(fencedCodeBlock);
```
Comment (yufeih) [inline src/docfx/lib/markdown/validation/DocsValidationExtension.cs:136]: consider call it `LineCount`

Comment (xichan-ms) [reply src/docfx/lib/markdown/validation/DocsValidationExtension.cs:136]: OK.

File: src/docfx/lib/markdown/validation/DocsValidationExtension.cs [hunk 2]
```diff
@@ -153,6 +154,21 @@ internal static class DocsValidationExtension
             }
         }
 
+        private static int GetFencedCodeBlockNetLineNumber(FencedCodeBlock fencedCodeBlock)
+        {
+            int netLineNum = 0;
+
+            for (int i = 0; i < fencedCodeBlock.Lines.Count; i++)
+            {
+                if (!string.IsNullOrWhiteSpace(fencedCodeBlock.Lines.Lines[i].ToString()))
```
Comment (OsmondJiang) [inline src/docfx/lib/markdown/validation/DocsValidationExtension.cs:163]: can we avoid ToString to determine whether it's empty？

Comment (xichan-ms) [reply src/docfx/lib/markdown/validation/DocsValidationExtension.cs:163]: I just confirm it, maybe not.

**Comment by xichan-ms** [comment]
Diff Explanations:
(1) Some warnings disappear.  **Expected.**
Example: line 141, https://github.com/MicrosoftDocs/sql-docs-pr/edit/master/docs/ado/guide/appendixes/microsoft-ole-db-provider-for-odbc.md
The linecount of this codeblock is 1, so do not execute the language missing validation.
(2) Some error message of warnings have changed. **Expected**.
Example: line 33, https://github.com/MicrosoftDocs/sql-docs-pr/edit/master/docs/ado/guide/data/accessing-rows-in-a-hierarchical-recordset.md
We have modify the error message. And the linecount of this codeblock is greater than 20, so execute the language missing validation.

File: src/docfx/lib/markdown/validation/DocsValidationExtension.cs (hunk 2 above)
Comment (yufeih) [reply src/docfx/lib/markdown/validation/DocsValidationExtension.cs:163]: `StringSlice.Trim` method should do the trick without calling `ToString`

Comment (xichan-ms) [reply src/docfx/lib/markdown/validation/DocsValidationExtension.cs]: OK. Got it.

File: src/docfx/lib/markdown/validation/DocsValidationExtension.cs [hunk 3]
```diff
@@ -153,6 +154,23 @@ internal static class DocsValidationExtension
             }
         }
 
+        private static int GetFencedCodeBlockNetLineCount(FencedCodeBlock fencedCodeBlock)
+        {
+            int netLineCount = 0;
+
+            for (int i = 0; i < fencedCodeBlock.Lines.Count; i++)
+            {
+                fencedCodeBlock.Lines.Lines[i].Slice.Trim();
```
Comment (OsmondJiang) [inline src/docfx/lib/markdown/validation/DocsValidationExtension.cs:163]: remove this line

**Pull request merged**